test/e2e/run.sh

```diff
@@ -874,6 +874,36 @@ $py_assertion
     return 0
 }
 
+fedora-set-kernel-cmdline() {
+    local e2e_defaults="$*"
+    vm-command "mkdir -p /etc/default; touch /etc/default/grub; sed -i '/e2e:fedora-set-kernel-cmdline/d' /etc/default/grub"
+    vm-command "echo 'GRUB_CMDLINE_LINUX_DEFAULT=\"\${GRUB_CMDLINE_LINUX_DEFAULT} ${e2e_defaults}\"' >> /etc/default/grub" || {
+        command-error "writing new command line parameters failed"
+    }
+    vm-command "grub2-mkconfig -o /boot/grub2/grub.cfg" || {
+        command-error "updating grub failed"
+    }
+}
+
+ubuntu-set-kernel-cmdline() {
+    local e2e_defaults="$1"
+    vm-command "echo 'GRUB_CMDLINE_LINUX_DEFAULT=\"\${GRUB_CMDLINE_LINUX_DEFAULT} ${e2e_defaults}\"' > /etc/default/grub.d/60-e2e-defaults.cfg" || {
+        command-error "writing new command line parameters failed"
+    }
+    vm-command "update-grub" || {
+        command-error "updating grub failed"
+    }
+}
+
+vm-reboot() { # script API
+    # Usage: vm-reboot
+    #
+    # Reboots the virtual machine and waits that the ssh server starts
+    # responding again.
+    vm-command "reboot"
+    sleep 5
```

Comment:
The caller of `vm-reboot` expects in this test (and it's safe to assume in other tests, too) that the vm is up and running and ready for next commands immediately when this function returns. 

In cri-resource-manager we had a function where we (on the host) wait that vm responds. If not, it's better to error out with clear error message than let the test script continue and fail later when the root cause is harder to debug.

See:
https://github.com/intel/cri-resource-manager/blob/master/demo/lib/vm.bash#L622

(You don't need to copy vm-monitor-based forced reboot here, but waiting for ssh connection to be available is a good idea.)

Reply:
I have tried but it seems that the `host-wait-vm-ssh-server` function is broken. Do you prefer that function to be here or is it fine to have my own custom func?

Reply:
Is there something else broken than `vm-monitor`-based hard reset? I think that hard reset should be dropped anyway. This function should be about waiting for the ssh server to respond and not really doing any other tricks but waiting. I was hoping that you could add the function ((possibly directly from cri-resource-manager and just ripping off the vm-monitor part).

If there is something wrong with it, then it needs to be implemented in some other way. We'll anyway need some way to wait for virtual machines to be again accessible through ssh after reboots. Something that continues pretty soon when ssh works again, but can wait for some time for the vm to boot. That is, something better than `sleep 5` or `sleep 5m` and just wishing that the vm woke up.

Reply:
The reason why `host-wait-vm-ssh-server` can't be lifted over as such from CRI Resource Manager e2e tests is that we now use vagrant instead of govm for test VMs. One stupidly simplistic way to update a `host-wait-vm-ssh-server` for the way we now use vagrant would be to have something like this:

```
host-wait-vm-ssh-server() {
    local _timeout=${timeout:-10}
    local _waited

    _waited=0; while [[ $_waited < $_timeout ]]; do
        if [ ! -e $OUTPUT_DIR/vagrant/.ssh ]]; then
            sleep 1
        else
            ssh -o ConnectTimeout=1 -F $OUTPUT_DIR/vagrant/.ssh-config true
            if [ $? == 0 ]; then
                return 0
            fi
        fi
        _waited=$((_waited + 1))
    done
}
```

Reply:
Tiny fix: instead of `while [[ ... ]]` use `while (( ... ))`, because `[[ 2 < 10 ]]` is false (due to lexicographic order) but `(( 2 < 10 ))` is true.